src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -326,70 +333,106 @@ export function openFacilityPageV2(page, uiSchema, schema) {
-      let locations = null;
-      let locationId = data.vaFacility;
+      let directFacilityIds = [];
+      let requestFacilityIds = [];
 
       dispatch({
         type: FORM_PAGE_FACILITY_V2_OPEN,
       });
 
-      try {
-        if (!parentFacilities) {
-          dispatch({ type: FORM_FETCH_PARENT_FACILITIES });
-          parentFacilities = await getOrganizations({
-            siteIds: userSiteIds,
-            useVSP,
-          });
-          dispatch({
-            type: FORM_FETCH_PARENT_FACILITIES_SUCCEEDED,
-            parentFacilities,
+      // Fetch parent facilities if we haven't already in
+      // checkCommunityCareEligibility()
+      if (!parentFacilities) {
+        parentFacilities = await getOrganizations({
+          siteIds: userSiteIds,
+          useVSP,
+        });
+      }
+
+      // Fetch direct and request eligible facilities and their details
+      // if we haven't already
+      if (!typeOfCareFacilities) {
+        const criteria = await Promise.all([
+          getDirectBookingEligibilityCriteria(userSiteIds),
+          getRequestEligibilityCriteria(userSiteIds),
+        ]);
+
+        // Fetch facilities that support direct scheduling and filter
+        // only those that support the selected type of care
+        directFacilityIds = criteria[0]
+          ?.filter(facility =>
+            facility?.coreSettings?.some(
+              setting => setting.id === typeOfCareId,
+            ),
+          )
+          ?.map(facility => facility.id);
+
+        // Fetch facilities that support requests and filter
+        // only those that support the selected type of care
+        requestFacilityIds = criteria[1]
+          ?.filter(facility =>
+            facility?.requestSettings?.some(
+              setting => setting.id === typeOfCareId,
+            ),
+          )
+          ?.map(facility => facility.id);
+
+        const uniqueIds = Array.from(
+          new Set([...directFacilityIds, ...requestFacilityIds]),
+        );
+
+        // The above API calls only return the ids. Make an additional
+        // call to getLocations so we can get additional details such
+        // as name, address, coordinates, etc.
+        if (uniqueIds.length) {
+          typeOfCareFacilities = await getLocations({
+            facilityIds: uniqueIds,
           });
+
+          // Update the retrieved locations with requestSupported and
+          // directSchedulingSupported, as well as replace IDs for dev/staging
+          typeOfCareFacilities = typeOfCareFacilities?.map(location =>
+            setSupportedSchedulingMethods({
+              location,
+              requestFacilityIds,
+              directFacilityIds,
+            }),
+          );
         }
-      } catch (err) {
-        dispatch({ type: FORM_FETCH_PARENT_FACILITIES_FAILED });
       }
 
-      locations = newAppointment.facilities[typeOfCareId] || null;
+      const canShowFacilities = !!parentId || parentFacilities?.length === 1;
```

Comment:
I don't think you need this logic to get the siteId, it's the first three digits of whatever facility id is chosen (excluding the `var` prefix).

Reply:
Gotcha.  Created a helper function called `getSiteIdFromFakeFHIRId()` to do this which I've added to to the Location service.